### PR TITLE
fix #209 getAllByIndex not resolved

### LINKED ISSUE #209: getAllByIndex observable won't resolve if cursor has value
I have an objectStore 'activity' with index 'company_id'
Insert 2 entries with same 'company_id' say 'id1'

getAllByIndex('activity', 'company_id', IDBKeyRange.only(id1').subscribe((result) => console.log(result));

This observable will never be resolved. I'm wondering we are not listening to cursor onsuccess event for processing subsequent records / get notified when the cursor reaches the end.

`    getAllByIndex(storeName, indexName, keyRange) {
        const data = [];
        return from(new Promise((resolve, reject) => {
            this.openCursorByIndex(storeName, indexName, keyRange)
                .pipe(take(1))
                .subscribe((event) => {
                const cursor = event.target.result;
                if (cursor) {
                    data.push(cursor.value);
                    cursor.continue();
                }
                else {
                    resolve(data);
                }
            }, (reason) => reject(reason));
        }));
    }`

### PR DESCRIPTION
**Pull request by desmondchou**
this commit is to fix #209 which cursor is not iterated and resolve result.

### CODE REVIEW

**Review by assuncaocharles — APPROVED** [review]

**Review by assuncaocharles — COMMENTED** [review]
Hey @desmondchou 

Can you add some test for it?

File: projects/ngx-indexed-db/src/lib/ngx-indexed-db.service.ts [hunk 1]
```diff
@@ -342,25 +342,27 @@ export class NgxIndexedDBService<T = any> {
...
-                cursor.continue();
-              } else {
-                resolve(data);
-              }
-            },
-            (reason) => reject(reason)
-          );
-      })
-    );
+    const data = [];
```
Comment (assuncaocharles) [inline projects/ngx-indexed-db/src/lib/ngx-indexed-db.service.ts]:
```suggestion
    const data: T[] = [];
```

File: projects/ngx-indexed-db/src/lib/ngx-indexed-db.service.ts [hunk 2]
```diff
@@ -343,24 +343,26 @@ export class NgxIndexedDBService<T = any> {
...
-                cursor.continue();
-              } else {
-                resolve(data);
-              }
-            },
-            (reason) => reject(reason)
-          );
-      })
-    );
+    return from(new Promise((resolve, reject) => {
```
Comment (assuncaocharles) [inline projects/ngx-indexed-db/src/lib/ngx-indexed-db.service.ts]:
```suggestion
    return from(new Promise<T[]>((resolve, reject) => {
```

File: projects/ngx-indexed-db/src/lib/ngx-indexed-db.service.ts [hunk 3]
```diff
@@ -343,24 +343,26 @@ export class NgxIndexedDBService<T = any> {
...
+    return from(new Promise<T[]>((resolve, reject) => {
+      openDatabase(this.indexedDB, this.dbConfig.name, this.dbConfig.version)
+      .then((db) => {
+      validateBeforeTransaction(db, storeName, reject);
+      const transaction = createTransaction(db, optionsGenerator(DBMode.readonly, storeName, reject, resolve));
+      const objectStore = transaction.objectStore(storeName);
+      const index = objectStore.index(indexName);
+      const request = index.openCursor(keyRange);
+      request.onsuccess = (event) => {
+          const cursor = event.target.result;
```
Comment (assuncaocharles) [inline projects/ngx-indexed-db/src/lib/ngx-indexed-db.service.ts]:
```suggestion
          const cursor: IDBCursorWithValue = (event.target as IDBRequest<IDBCursorWithValue>).result;
```

**Comment by desmondchou** [comment]
> Hey @desmondchou
> 
> Can you add some test for it?

I'm new to these. Would you mind showing me any file which I can reference for test adding? Also, thanks for fixing the type check error of event.target.result.

**Comment by assuncaocharles** [comment]
> > Hey @desmondchou
> > Can you add some test for it?
> 
> I'm new to these. Would you mind showing me any file which I can reference for test adding? Also, thanks for fixing the type check error of event.target.result.

For now you can only go for the playground and add the usage as you would use in a normal application

**Review by assuncaocharles — APPROVED** [review]

**Comment by assuncaocharles** [comment]
Thank you @desmondchou

**Pull request merged**